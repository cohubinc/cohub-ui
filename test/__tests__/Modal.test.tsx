import React from "react";
import { mount } from "enzyme";

import { Modal, StoryHelpers } from "dist";
import { render, fireEvent } from "@testing-library/react";

describe("Modal component", () => {
  it("renders", async () => {
    const wrapper = mount(
      <Modal open={true} onClose={() => undefined}>
        <span>Content</span>
      </Modal>
    );

    expect(wrapper).toMatchSnapshot();
  });

  it("toggles on click", async () => {
    const modalId = "modal-id-toggles-on-click";
    const closeButtonId = "close-toggles-on-click";
    const { getByTestId } = render(
      <StoryHelpers.StateContainer defaultState={true}>
        {({ state, setState }) => (
          <Modal open={state as boolean} onClose={() => setState(false)}>
            <div data-testid={modalId}>
              <div>Modal</div>
              <button
                data-testid={closeButtonId}
                onClick={() => setState(false)}
              >
                Close
              </button>
            </div>
          </Modal>
        )}
      </StoryHelpers.StateContainer>
    );
    const modal = getByTestId(modalId);
    const closeButton = getByTestId(closeButtonId);

    expect(modal).toBeVisible();
    fireEvent.click(closeButton);
    expect(modal).not.toBeVisible();
  });
});
