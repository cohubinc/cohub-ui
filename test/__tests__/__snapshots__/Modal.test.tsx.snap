// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component renders 1`] = `
<Modal
  onClose={[Function]}
  open={true}
>
  <Backdrop
    containerClass="CohubModal"
    focusTrapped={true}
    onClose={[Function]}
    open={true}
    showCloseIcon={true}
  >
    <Modal
      animationDuration={500}
      blockScroll={true}
      center={false}
      classNames={
        Object {
          "closeButton": "closeButton",
          "modal": "modal",
          "overlay": "CohubBackdrop CohubModal",
        }
      }
      classes={
        Object {
          "closeButton": "styles_closeButton__20ID4",
          "closeIcon": "styles_closeIcon__1QwbI",
          "modal": "styles_modal__gNwvD",
          "modalCenter": "styles_modalCenter__L9F2w",
          "overlay": "styles_overlay__CLSq-",
          "transitionEnter": "styles_transitionEnter__3j_-a",
          "transitionEnterActive": "styles_transitionEnterActive___eQs7",
          "transitionExit": "styles_transitionExit__1KmEf",
          "transitionExitActive": "styles_transitionExitActive__1nQXw",
        }
      }
      closeIconSize={28}
      closeIconSvgPath={
        <svg
          fill="none"
          height={44}
          viewBox="0 0 24 24"
          width={44}
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 0.974332L11.025 0L6.00034 5.02532L0.975021 0L0 0.974332L5.02532 5.99966L0 11.025L0.975021 11.9993L6.00034 6.97399L11.025 11.9993L12 11.025L6.97468 5.99966L12 0.974332Z"
            fill="hsl(0, 0%, 100%)"
          />
        </svg>
      }
      closeOnEsc={true}
      closeOnOverlayClick={true}
      focusTrapOptions={Object {}}
      focusTrapped={true}
      onClose={[Function]}
      onEscKeyDown={[Function]}
      onOverlayClick={[Function]}
      open={true}
      role="dialog"
      showCloseIcon={true}
      styles={
        Object {
          "overlay": undefined,
        }
      }
    >
      <Portal
        containerInfo={
          <div>
            <div
              class="styles_overlay__CLSq- CohubBackdrop CohubModal styles_transitionEnter__3j_-a styles_transitionEnterActive___eQs7"
            >
              <div
                aria-modal="true"
                class="styles_modal__gNwvD modal"
                role="dialog"
              >
                <div>
                  <div
                    class="modalBody undefined"
                    style="width: 360px;"
                  >
                    <div
                      class="Segment-module_CohubSegment__3MMwJ Segment-module_padded__39Fvk "
                      style="box-shadow: 0px 11px 15px rgba(0, 0, 0, 0.2), 0px 9px 46px rgba(0, 0, 0, 0.12), 0px 24px 38px rgba(0, 0, 0, 0.14); background-color: hsl(0, 0%, 100%); width: 100%;"
                    >
                      <span>
                        Content
                      </span>
                    </div>
                  </div>
                  <button
                    class="styles_closeButton__20ID4 closeButton"
                  >
                    <svg
                      class="styles_closeIcon__1QwbI"
                      height="28"
                      viewBox="0 0 36 36"
                      width="28"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <svg
                        fill="none"
                        height="44"
                        viewBox="0 0 24 24"
                        width="44"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M12 0.974332L11.025 0L6.00034 5.02532L0.975021 0L0 0.974332L5.02532 5.99966L0 11.025L0.975021 11.9993L6.00034 6.97399L11.025 11.9993L12 11.025L6.97468 5.99966L12 0.974332Z"
                          fill="hsl(0, 0%, 100%)"
                        />
                      </svg>
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        }
      >
        <CSSTransition
          appear={true}
          classNames={
            Object {
              "appear": "styles_transitionEnter__3j_-a",
              "appearActive": "styles_transitionEnterActive___eQs7",
              "enter": "styles_transitionEnter__3j_-a",
              "enterActive": "styles_transitionEnterActive___eQs7",
              "exit": "styles_transitionExit__1KmEf",
              "exitActive": "styles_transitionExitActive__1nQXw",
            }
          }
          in={true}
          onEntered={[Function]}
          onExited={[Function]}
          timeout={500}
        >
          <Transition
            appear={true}
            enter={true}
            exit={true}
            in={true}
            mountOnEnter={false}
            onEnter={[Function]}
            onEntered={[Function]}
            onEntering={[Function]}
            onExit={[Function]}
            onExited={[Function]}
            onExiting={[Function]}
            timeout={500}
            unmountOnExit={false}
          >
            <div
              className="styles_overlay__CLSq- CohubBackdrop CohubModal"
              onClick={[Function]}
            >
              <div
                aria-modal="true"
                className="styles_modal__gNwvD modal"
                onClick={[Function]}
                onMouseDown={[Function]}
                onMouseUp={[Function]}
                role="dialog"
              >
                <FocusTrap
                  _createFocusTrap={[Function]}
                  active={true}
                  focusTrapOptions={
                    Object {
                      "clickOutsideDeactivates": true,
                    }
                  }
                  paused={false}
                  tag="div"
                >
                  <div>
                    <div
                      className="modalBody undefined"
                      hidden={false}
                      style={
                        Object {
                          "width": "360px",
                        }
                      }
                    >
                      <Segment
                        bordered={false}
                        className=""
                        elevation={24}
                        padded={true}
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      >
                        <div
                          className="Segment-module_CohubSegment__3MMwJ Segment-module_padded__39Fvk "
                          style={
                            Object {
                              "backgroundColor": "hsl(0, 0%, 100%)",
                              "border": "",
                              "boxShadow": "0px 11px 15px rgba(0, 0, 0, 0.2), 0px 9px 46px rgba(0, 0, 0, 0.12), 0px 24px 38px rgba(0, 0, 0, 0.14)",
                              "width": "100%",
                            }
                          }
                        >
                          <span>
                            Content
                          </span>
                        </div>
                      </Segment>
                    </div>
                    <CloseIcon
                      classNames={
                        Object {
                          "closeButton": "closeButton",
                          "modal": "modal",
                          "overlay": "CohubBackdrop CohubModal",
                        }
                      }
                      classes={
                        Object {
                          "closeButton": "styles_closeButton__20ID4",
                          "closeIcon": "styles_closeIcon__1QwbI",
                          "modal": "styles_modal__gNwvD",
                          "modalCenter": "styles_modalCenter__L9F2w",
                          "overlay": "styles_overlay__CLSq-",
                          "transitionEnter": "styles_transitionEnter__3j_-a",
                          "transitionEnterActive": "styles_transitionEnterActive___eQs7",
                          "transitionExit": "styles_transitionExit__1KmEf",
                          "transitionExitActive": "styles_transitionExitActive__1nQXw",
                        }
                      }
                      closeIconSize={28}
                      closeIconSvgPath={
                        <svg
                          fill="none"
                          height={44}
                          viewBox="0 0 24 24"
                          width={44}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M12 0.974332L11.025 0L6.00034 5.02532L0.975021 0L0 0.974332L5.02532 5.99966L0 11.025L0.975021 11.9993L6.00034 6.97399L11.025 11.9993L12 11.025L6.97468 5.99966L12 0.974332Z"
                            fill="hsl(0, 0%, 100%)"
                          />
                        </svg>
                      }
                      id={null}
                      onClickCloseIcon={[Function]}
                      styles={
                        Object {
                          "overlay": undefined,
                        }
                      }
                    >
                      <button
                        className="styles_closeButton__20ID4 closeButton"
                        id={null}
                        onClick={[Function]}
                      >
                        <svg
                          className="styles_closeIcon__1QwbI"
                          height={28}
                          viewBox="0 0 36 36"
                          width={28}
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <svg
                            fill="none"
                            height={44}
                            viewBox="0 0 24 24"
                            width={44}
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M12 0.974332L11.025 0L6.00034 5.02532L0.975021 0L0 0.974332L5.02532 5.99966L0 11.025L0.975021 11.9993L6.00034 6.97399L11.025 11.9993L12 11.025L6.97468 5.99966L12 0.974332Z"
                              fill="hsl(0, 0%, 100%)"
                            />
                          </svg>
                        </svg>
                      </button>
                    </CloseIcon>
                  </div>
                </FocusTrap>
              </div>
            </div>
          </Transition>
        </CSSTransition>
      </Portal>
    </Modal>
  </Backdrop>
</Modal>
`;
